meson.build

```diff
@@ -584,6 +584,31 @@ if get_option('build_backends')
     has_backends = true
   endif
 
+  ## ~~~~~~~~
+  ## Metal
+  ## ~~~~~~~~
+  # Metal backend only available on MacOS.
+  if host_machine.system() == 'darwin' and get_option('metal')
```

Comment:
I think you can move the `metal_frameworks =...` bit earlier and then replace the `host_machine.system() == 'darwin'` with `metal_frameworks.found()` as according to the meson docs the `appleframeworks` approach will not find dependencies on other systems. Then you can skip the nested if and do the `add_languages()` unconditionally.

Reply:
It was primarily because I didn't want to kick off a framework search if `get_option('metal')` is `false`, that I accepted going with the nested if. If that's not much of a problem, then we can go with your proposal. Especially, if we keep `metal=true` as the default in `meson.options`

Reply:
I don't see why we wouldn't have `metal=true` by default. And I expect the framework search is very fast on other systems so no point complicating things.

Reply:
It short-circuits on any compiler other than clang, with a FIXME that it should use appleclang only.

Then it short-circuits again, if `clang -v -E -` doesn't report any framework directories.

Reply:
BTW you may want to use feature options instead :)
- `-Dmetal=auto`: equivalent to `required: false`
- `-Dmetal=enabled`: equivalent to `required: true`
- `-Dmetal=disabled`: mostly equivalent to `-Dmetal=false` if and only if that uses an `if get_option('metal')` block to skip invoking `dependency()` at all, although the disabled feature also logs e.g. "dependency AppleFrameworks (modules: ...) skipped: feature metal is disabled"

Reply:
So we would do
```
 metal_frameworks = dependency('appleframeworks', 
                               modules : ['Foundation', 'Metal', 'MetalPerformanceShaders'],
                               required: get_option(metal))
if metal_frameworks.found() and add_languages('objc', 'objcpp') ...
```
and
```
option('metal',
       type: 'feature',
       value: `auto`,
       description: 'Enable Metal backend')
```
Is this correct?

Reply:
That looks correct, yes.